pkg/api/persistentvolume/util.go

```diff
@@ -111,6 +111,33 @@ func VisitPVSecretNames(pv *api.PersistentVolume, visitor Visitor) bool {
 		if source.StorageOS.SecretRef != nil && !visitor(source.StorageOS.SecretRef.Namespace, source.StorageOS.SecretRef.Name) {
 			return false
 		}
+	case source.CSI != nil:
+		switch {
+		case source.CSI.ControllerPublishSecretRef != nil:
```

Comment:
this shouldn't be a switch statement. more than one of these can be set at a time, and you'd want all to be visited

Reply:
If I use fallthrough then?

Reply:
no, just a series of checks:

```go
case source.CSI != nil:
  if source.CSI.ControllerPublishSecretRef != nil {
    if !visitor(source.CSI.ControllerPublishSecretRef.Namespace, source.CSI.ControllerPublishSecretRef.Name) {
      return false
    }
  }

  if source.CSI.NodePublishSecretRef != nil {
    if !visitor(source.CSI.NodePublishSecretRef.Namespace, source.CSI.NodePublishSecretRef.Name) {
      return false
    }
  }

  if source.CSI.NodeStageSecretRef != nil {
    if !visitor(source.CSI.NodeStageSecretRef.Namespace, source.CSI.NodeStageSecretRef.Name) {
      return false
    }
  }
```